diff -aurN rpm-4.5.orig/scripts/pkgconfigdeps.sh rpm-4.5/scripts/pkgconfigdeps.sh
--- rpm-4.5.orig/scripts/pkgconfigdeps.sh	2012-12-30 11:36:51.204543929 +0100
+++ rpm-4.5/scripts/pkgconfigdeps.sh	2012-12-27 21:26:20.000000000 +0100
@@ -16,12 +16,19 @@
     while read filename ; do
     case "${filename}" in
     *.pc)
-	# Query the dependencies of the package.
-	$pkgconfig --print-provides "$filename" 2> /dev/null | while read n r v ; do
-	    # We have a dependency.  Make a note that we need the pkgconfig
-	    # tool for this package.
-	    echo "pkgconfig($n)" "$r" "$v"
-	done
+	if [[ "$(dirname ${filename})" =~ pkgconfig ]]; then
+		DIR=`dirname ${filename}`
+		PKG_CONFIG_PATH="$DIR:$DIR/../../share/pkgconfig"
+		export PKG_CONFIG_PATH
+		$pkgconfig --print-provides "$filename" 2> /dev/null | while read n r v ; do
+			[ -n "$n" ] || continue
+		    	if  [ -n "$r" ] && [ -n "$v" ]; then
+				    echo "pkgconfig($n) $r $v"
+			    else
+				    echo "pkgconfig($n)"
+			fi
+		done
+    	fi
 	;;
     esac
     done
@@ -30,11 +37,20 @@
     while read filename ; do
     case "${filename}" in
     *.pc)
-	$pkgconfig --print-requires "$filename" 2> /dev/null | while read n r v ; do
-	    i="`expr $i + 1`"
-	    [ $i -eq 1 ] && echo "pkgconfig"
-	    echo "pkgconfig($n)" "$r" "$v"
-	done
+	if [[ "$(dirname ${filename})" =~ pkgconfig ]]; then
+		DIR=`dirname ${filename}`
+		PKG_CONFIG_PATH="$DIR:$DIR/../../share/pkgconfig"
+		export PKG_CONFIG_PATH
+		$pkgconfig --print-requires --print-requires-private "$filename" 2> /dev/null | while read n r v ; do
+			[ -n "$n" ] || continue
+		    	if  [ -n "$r" ] && [ -n "$v" ]; then
+				    echo "pkgconfig($n) $r $v"
+			    else
+				    echo "pkgconfig($n)"
+			fi
+		done
+    	fi
+	;;
     esac
     done
     ;;
